src/keboola_mcp_server/tools/_validate.py

```
@@ -0,0 +1,131 @@
+            Recovery instructions:
+            - Please check the json schema.
+            - Fix the errors in your input data to follow the schema.
+
+            Invalid input data:
+            {
+            "storage": {...}
+            }
+        """
+        str_repr = f'{super().__str__()}\n'
+        if self.initial_message:
+            str_repr += f'{self.initial_message}\n'
+        str_repr += f'{self._RECOVERY_INSTRUCTIONS}\n'
+        if self.invalid_json:
+            str_repr += f'\nInvalid input data:\n{json.dumps(self.invalid_json, indent=2)}\n'
+        return str_repr.rstrip()
+
+
+def validate_storage(storage: JsonDict, initial_message: Optional[str] = None) -> JsonDict:
+    """
+    Validate the storage configuration using jsonschema.
+
+    """
+    schema = _load_schema(ConfigurationSchemaResourceName.STORAGE)
+    expected_input_data = {'storage': storage.get('storage', storage)}
+    _validate_json_against_schema(
+        json_data=expected_input_data,
+        schema=schema,
+        initial_message=initial_message or STORAGE_VALIDATION_INITIAL_MESSAGE,
+    )
+    return storage
+
+
+def _validate_json_against_schema(
+    json_data: JsonDict, schema: dict[str, Any], initial_message: Optional[str] = None
+) -> bool:
+    """Validate JSON data against the provided schema."""
+    try:
+        jsonschema.validate(instance=json_data, schema=schema)
+        return True
+    except jsonschema.ValidationError as e:
+        raise RecoverableValidationError.create_from_values(e, invalid_json=json_data, initial_message=initial_message)
+    except jsonschema.SchemaError as e:
+        err_msg = (
+            f'The validation schema is not valid: {e} \n'
+            f'initial_message: {initial_message}\n'
+            f'schema: {schema}\n'
+            f'json_data: {json_data}'
+        )
+        LOG.error(f'{err_msg}')
+        raise jsonschema.SchemaError(f'{e}\n{err_msg}')  # this is not an Agent error, the schema is not valid
```

Umm, do we really want to fail the validation if the schema is invalid? This would effectively make it impossible for AI assistants to create/modify a component, if its developers failed to provide a valid configuration schema. And that is too restrictive, I think. I'd just log the exception (btw. it's better to use `LOG.exception()` to send the traceback in the log) and returned.